lib/dice/src/cert.rs

```diff
@@ -0,0 +1,346 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{
+    alias_cert_tmpl, deviceid_cert_tmpl, SerialNumber, NOTBEFORE_LENGTH,
+};
+use hubpack::SerializedSize;
+use salty::constants::{
+    PUBLICKEY_SERIALIZED_LENGTH, SIGNATURE_SERIALIZED_LENGTH,
+};
+use salty::signature::{Keypair, PublicKey};
+use serde::{Deserialize, Serialize};
+use serde_big_array::BigArray;
+use unwrap_lite::UnwrapLite;
+
+#[derive(Debug)]
+pub enum CertError {
+    BadSig,
+    NoPubKey,
+    NoSig,
+    NoSignData,
+    TooSmall,
+    NotFound,
+    NoCn,
+}
+
+pub trait Cert {
+    fn as_bytes(&self) -> &[u8];
+    fn as_mut_bytes(&mut self) -> &mut [u8];
+
+    const SERIAL_NUMBER_START: usize;
+    const SERIAL_NUMBER_END: usize;
+
+    fn get_serial_number(&self) -> u8 {
```

Comment:
Optional:

If you want to get fancy here, you could add a helper function of the form
```rust
    fn read_range<'a, T>(&'a self, r: core::ops::Range<usize>) -> T
    where T: TryFrom<&'a [u8]>
    {
        self.as_bytes()[r].try_into().unwrap_lite()
    }
```

Then, `get_serial_number` would become
```rust
fn get_serial_number(&self) -> u8 {
    u8::from_be_bytes(self.read_range(Self::SERIAL_NUMBER_RANGE))
}
```
(and similarly, the other getters would be a little less verbose)